aspnet/fundamentals/request-features.rst

```diff
@@ -1,8 +1,177 @@
-.. include:: /../common/stub-topic.txt
+Request Features
+================
 
-|stub-icon| Request Features
-============================
+By `Steve Smith`_
 
-.. include:: /../common/stub-notice.txt
+Individual web server features related to how HTTP requests and responses are handled have been factored into separate interfaces, defined in the `HttpAbstractions <https://github.com/aspnet/HttpAbstractions>`_ package. These abstractions are used by individual server implementations and middleware to create and modify the application's hosting pipeline.
 
-.. _issue: https://github.com/aspnet/Docs/issues/293
+In this article:
+	- `Feature interfaces`_
+	- `Feature collections`_
+	- `Middleware and request features`_
+	
+Feature interfaces
+------------------
+
+ASP.NET 5 defines a number of `Http Feature Interfaces <https://github.com/aspnet/HttpAbstractions/tree/dev/src/Microsoft.AspNet.Http.Features>`_, which are used by servers to identify which features they support. The most basic features of a web server are the ability to handle requests and return responses, as defined by the following feature interfaces:
+
+`IHttpRequestFeature <https://github.com/aspnet/HttpAbstractions/blob/dev/src/Microsoft.AspNet.Http.Features/IHttpRequestFeature.cs>`_
+	Defines the structure of an HTTP request, including the protocol, path, QueryString, headers, and body.
```

Comment:
with all the other parameters in this sentence. `query string` might fit better.